sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/JsonInferSchema.scala

```diff
@@ -79,7 +80,7 @@ private[sql] object JsonInferSchema {
 
   private[this] val structFieldComparator = new Comparator[StructField] {
     override def compare(o1: StructField, o2: StructField): Int = {
-      o1.name.compare(o2.name)
+      o1.name.compareTo(o2.name)
```

Comment:
`.compare` is a very expensive way of comparing two strings.

`compare`:
```
  public int compare(org.apache.spark.sql.types.StructField, org.apache.spark.sql.types.StructField);
    Code:
       0: new           #14                 // class scala/collection/immutable/StringOps
       3: dup
       4: getstatic     #20                 // Field scala/Predef$.MODULE$:Lscala/Predef$;
       7: aload_1
       8: invokevirtual #26                 // Method org/apache/spark/sql/types/StructField.name:()Ljava/lang/String;
      11: invokevirtual #30                 // Method scala/Predef$.augmentString:(Ljava/lang/String;)Ljava/lang/String;
      14: invokespecial #34                 // Method scala/collection/immutable/StringOps."<init>":(Ljava/lang/String;)V
      17: aload_2
      18: invokevirtual #26                 // Method org/apache/spark/sql/types/StructField.name:()Ljava/lang/String;
      21: invokevirtual #37                 // Method scala/collection/immutable/StringOps.compare:(Ljava/lang/String;)I
      24: ireturn
```

`compareTo`:
```
  public int compare(org.apache.spark.sql.types.StructField, org.apache.spark.sql.types.StructField);
    Code:
       0: aload_1
       1: invokevirtual #18                 // Method org/apache/spark/sql/types/StructField.name:()Ljava/lang/String;
       4: aload_2
       5: invokevirtual #18                 // Method org/apache/spark/sql/types/StructField.name:()Ljava/lang/String;
       8: invokevirtual #24                 // Method java/lang/String.compareTo:(Ljava/lang/String;)I
      11: ireturn
```